src/test/utils.test.ts

```diff
@@ -0,0 +1,93 @@
+import {buildAllPairs} from "../utils";
+
+test('buildAllPairs creates correlation pairs without missing any fields and without duplicates', () => {
+    const input = ["apple", "banana", "strawberries", "pears", "grapefruit", "bread", "milk"]
+    const results = buildAllPairs(input)
+    expect(results.length).toEqual(21);
+    expect(results).toEqual([
+        [
+          "apple",
+          "banana",
+        ],
+        [
+          "apple",
+          "strawberries",
+        ],
+        [
+          "apple",
+          "pears",
+        ],
+        [
+          "apple",
+          "grapefruit",
+        ],
+        [
+          "apple",
+          "bread",
+        ],
+        [
+          "apple",
+          "milk",
+        ],
+        [
+          "banana",
+          "strawberries",
+        ],
+        [
+          "banana",
+          "pears",
+        ],
+        [
+          "banana",
+          "grapefruit",
+        ],
+        [
+          "banana",
+          "bread",
+        ],
+        [
+          "banana",
+          "milk",
+        ],
+        [
+          "strawberries",
+          "pears",
+        ],
+        [
+          "strawberries",
+          "grapefruit",
+        ],
+        [
+          "strawberries",
+          "bread",
+        ],
+        [
+          "strawberries",
+          "milk",
+        ],
+        [
+          "pears",
+          "grapefruit",
+        ],
+        [
+          "pears",
+          "bread",
+        ],
+        [
+          "pears",
+          "milk",
+        ],
+        [
+          "grapefruit",
+          "bread",
+        ],
+        [
+          "grapefruit",
+          "milk",
+        ],
+        [
+          "bread",
+          "milk",
+        ],
```

Comment:
For these sorts of results, I really think approval testing is better. You don't need to hardcode the expected outcome, which saves alot of time when you want to change the input

Reply:
What do you mean by approval testing?

Reply:
We may benefit from having hardcoded test just to make sure that it's doing exactly what we expect it to do and then being more generic in other tests we may add here.

Reply:
Approval testing is different in that you don't supply an expected result for the first test. You simply run the test, and manually confirm if the output is correct. If so, you _approve_ it as the correct output, and that becomes the expected result

Reply:
But I get you, this works for now

Reply:
Oh interesting! Yeah, we can add some of those down the line. It'll be good to have a mix. The non-approval stuff, we can use Github actions to run automatically at PR creation.